### ISSUE TITLE
Proper Attribution and Licensing

### ISSUE THREAD

**Issue by nadimkobeissi** [ioerror/mpOTR#2]
Hi Jacob,
You've copy-pasted [Cryptocat's mpOTR research](https://github.com/cryptocat/cryptocat/wiki/mpOTR-Specification) from our project wiki without asking the Cryptocat Project for permission and without crediting any author other than the implied credit to yourself.

Please, at the very least have the courtesy to ask for permission to copy research from the Cryptocat Project, and at the very least credit those who've worked on those research so that you're not accused of plagiarism. I've already touched upon this subject [here](https://github.com/cryptocat/cryptocat/issues/82#issuecomment-13577097) but your response has been despondent to say the very least.

**Comment by ioerror** [comment]
I am one of the original authors of the document. When it was removed from cryptocat's git, the previous git history associated with the document was essentially lost. That is - the stuff that was ever in git to begin with - not much, sadly.

The document is a living document and if you have a list of names that you feel should be on the document, I've suggested making a pull request. If the document was fine to have on your wiki as was, I hardly see how it is reasonable to ding me for having a copy of the same document that helped to write in the first place.

**Comment by nadimkobeissi** [comment]
You've closed the issue, do you consider this problem resolved?

**Comment by ioerror** [comment]
I consider my statement above to be clear - send a pull request with the information that you feel is missing and I'll merge it. If you're here to accuse me of plagiarism, I'll ask where in your history my name was credited for my labor on this document.

It isn't finished and i intend to talk to each of the people involved in helping to author it as some may or may not wish to have their name on the document.

If you want to resolve the problem, submit a pull request or suggest some names that are missing from the document. If you just want to pick a fight, keep refusing to work towards a solution Nadim.

**Comment by nadimkobeissi** [comment]
Jacob,
You do not have permission to copy this document. Please cease stealing our research. Your contributions to this document were done with you being part of the Cryptocat team at the WSJ hack-a-thon and thus this document belongs to the Cryptocat Project. You have never asked for credit — I would be happy to credit you if you wish.

**Comment by ioerror** [comment]
Huh?

![Screenshot-Twitter - kaepora: Finally some initiative on - Google Chrome (https://f.cloud.github.com/assets/145203/159194/e2853760-7709-11e2-9e37-db96195b6b0b.png)

I never signed a copyright assignment document. I am an original author of the document. Furthermore, it was emailed by many people to the otr-dev mailing list.

In any case, are you seriously asserting that you have proprietary files in your free software project and that original authors lose their copyright when you arbitrarily put the document into your wiki?

Seriously? You don't have better things to do with your time?

**Comment by nadimkobeissi** [comment]
That tweet does not count as any sort of sanctioned permission whatosever. It was written during my first request that you ask for permission, and before I realized you were avoiding my requests that you ask us for permission and that you credit the project.

You did not sign a copyright assignment, but this research is a joint effort within the Cryptocat Project and you are currently plagiarizing it without permission or attribution, from or towards, any of the other authors or the underlying software project.

Furthermore, you have repeatedly demonstrated bad faith during this plea and I will begin drafting a letter towards achieving fairness in this situation.

**Comment by ioerror** [comment]
Your wiki shows only yourself as the author of the document. You're being unreasonable and if you feel that people should be specifically credited, I encourage you to list their names. I've started to go through the lost git history to ensure that people are actually credited, something that you neglected to do for nearly nine months, I might add.

Please stop being so difficult, if you actually wanted to resolve the conflict you would take actions to resolve it. You don't own the document and I am not preventing you from doing anything with the version you are hosting.

**Comment by nadimkobeissi** [comment]
Thank you for undertaking your duty to list the proper authors of the research.

Cryptocat's wiki version in fact also lists Abel as a contributor in the wiki edit history. I cannot edit your repository to put credits in, but you have been able to edit the wiki to attribute yourself for as long as it has existed. No one had asked for attribution, including you.

You have copy-pasted research jointly written for the Cryptocat Project without permission. You have no permission to do this. Please ask for permission and please properly credit the authors.

**Comment by ioerror** [comment]
I've pushed commits that add the names that you lost when you moved from git to the Wiki. I believe I have added all of the names of people who mentioned being comfortable with their names on such a document.

I don't have to ask you for permission to use a document that I co-authored. We never had an agreement that made you the sole owner of the document.

If you won't stop being unreasonable, I'll ask you to stop hosting documents that I co-authored when you seek to limit derivative works by other co-authors. A lack of copyright assignment agreements will be more harmful to your project than to this git repository.

**Comment by dgoulet** [comment]
I don't want to put oil on the fire but the Cryptocat Project seems to be licensed as AGPL. As long as this document stays that way in this git repo. and the authors copyright are kept, there is no need for permission to fork this document. I do agree in terms of collaboration that duplicates can be a bad idea but I guess finding a solution to that would be easier than arguing on who has the rights.

My two cents. :)

**Comment by nadimkobeissi** [comment]
I think your crediting list is complete, thank you. This is the first act of good faith I've seen from you since the beginning of this conflict.

If Jacob is willing to clarify the license of this publication as AGPL, then I will back off.

**Comment by ioerror** [comment]
Nadim - Are you asserting that you wish for your contributions to this document to be under the AGPL?

**Comment by nadimkobeissi** [comment]
Jacob — yes, that was the original intended license of the document. However, you have:
- Not included any licenses.
- Failed to attribute the original licensees.

**Comment by ioerror** [comment]
Your contributions are noted as AGPL - thanks. I'll make a note of it in the document.

**Comment by nadimkobeissi** [comment]
This entire document is licensed as AGPL, not just my contributions! You took it from an AGPL repository!

**Comment by ioerror** [comment]
I don't remember agreeing to license it under the AGPL. While I'm also happy to add my contributions under the AGPL, I'm not comfortable declaring licensing without talking to the other authors explicitly.

Did you get copyright assignment from the other authors before you put it into the git repo?

**Comment by nadimkobeissi** [comment]
Jacob,
You specifically asked me to change Cryptocat's license from Creative Commons to AGPL3 in April 2012. You were the main person who asked me to do this. @hellais was present and can testify.

In terms of talking to other authors explicitly, you have already copied their research without permission. You are acting in bad faith and it's obvious to everyone.

**Comment by ioerror** [comment]
This file is a document edited by multiple people over many computers, networks, git repos, wikis and so on. The license of CryptoCat is irrelevant to the discussion - you don't have the right to license my contributions as you see fit simply because you put a text file in your git repository.

You have declared your contributions to be under the AGPL. That said, I'm happy to have the document under the AGPL but I refuse to speak for the other authors.

I would prefer that they all weigh in on the license you have selected and that we give them an opportunity to ensure that we aren't changing their expectations. As an example, I was surprised that all history had been lost in your transition from git to wiki page and that it is now declared a CryptoCat document under the AGPL. I'm fine with the AGPL part now that we've discussed it. However, I am not fine with the part where you did not credit the authors and declared the entire work property of the CryptoCat project, and then proceeded to start a large argument across many email lists, twitter and on git.

You could have just opened a pull request and I would have merged your proposed changed. Just as i did with another suggested fix you made moments ago.

**Comment by nadimkobeissi** [comment]
> you don't have the right to license my contributions as you see fit simply because you put a text file in your git repository.

This is **exactly** how licenses are stored. This license was implemented, _under your recommendation_, across the entire Cryptocat Project repository, including its wiki.

This license has been there visible for all to see should they contribute to their project. The contributors have been submitting content under the terms of the AGPLv3, which is clearly referenced in the Cryptocat project repository and README, as well as our project website.

I will immediately propose a pull request for the AGPLv3 licensing, and I expect it to be accepted without any delay. I shouldn't be making these kinds of pull requests in the first place. They are your responsibility and your lack of performing this kind of basic responsibility is purely reprehensible bad faith.

**Comment by nadimkobeissi** [comment]
[My pull request is now available.](https://github.com/ioerror/mpOTR/pull/4)

**Comment by ioerror** [comment]
I understand that this is how licenses are stored. How exactly do you have the right to put this document into git, then entirely un-credited into the wiki and then to declare it AGPL without respecting the rights of the other authors?

I'll gladly accept a pull request for a LICENSE file of the AGPL but again, I'd like _all_ of the original authors to agree to that license explicitly before we declare it to be the case without issue. This will avoid all conflict in the future.

You are declaring it for everyone and while I'm amenable to your choice, we haven't spoken to the other authors. We can easily add a sign-off to the document that ensures that the document is freely licensed. Until everyone has their say, we'll ensure that no one feels like they didn't have a chance or that someone else spoke for someone else unreasonably.

I think my proposed solution of giving everyone a chance to decide on AGPL is reasonable and that we don't even disagree on the LICENSING topic.

**Comment by nadimkobeissi** [comment]
> How exactly do you have the right to put this document into git, then entirely un-credited into the wiki and then to declare it AGPL without respecting the rights of the other authors?

Cryptocat was a licensed project long before any of the relevant authors contributed to it, and the license was made clearly visible on the repository which host the code, the documentation and the wiki. **May I remind you that you yourself proposed the license change to AGPLv3.**

If you wish to speak to the other authors, you may, but you are, until then, in violation of the Cryptocat Project's AGPLv3 license unless you accept my pull request. If you cannot accept my pull request, you remain in violation of our license and must close this repository without delay.

**Comment by ioerror** [comment]
I added the AGPL license and a note about future licensing.

**Comment by fwalgur** [comment]
Nadim,

You frequently symbolize a whiny child. You as well, obviously exert an extensive mental health issue. Not to mention your recent media ploy regarding surveillance by Canadian government which was invented mentally to satisfy your thirst for concentration.

Seek help ASAP.

**Comment by ioerror** [comment]
@fwalgur please try to keep things civil. :(

**Comment by nadimkobeissi** [comment]
@ioerror There's a problem with this part of your commit:

> Nadim Kobeissi has asserted that this document is licensed under the AGPL without discussing with any of the other authors. This may or may not actually count as a LICENSE declaration.

Actually, since the project was already AGPLv3 and the license was properly displayed at the time of contribution, there is no need to discuss it with other authors since due diligence has already been made to make the license available. This paragraph is therefore inaccurate.

**Comment by nadimkobeissi** [comment]
You may also want to consider my latest pull request (#5) which resolves this issue and also adds the license reference to the actual document.

**Comment by ioerror** [comment]
You're debating about the project's license. I do not dispute the license of Cryptocat. 

However, you have no right to assert total ownership of a derivative work. I have freely agreed to license my contributions under the AGPL. I am not willing to assert that for other people without consulting with them.

**Comment by nadimkobeissi** [comment]
I am not asserting total ownership of a derivative work. I am asserting that the derivative work must be released under a similar license in the vein of GPL and proper free software practice and attribution.

I will email the other respective authors and ask them to participate in this thread. In the meantime, please take a look at #5. Your commit still does not mention any license in the main document.

**Comment by nadimkobeissi** [comment]
@ioerror I have emailed the other authors (or at least, the ones I _could_ email.) In the meantime. I urge you to recognize that all contributions have in fact fairly been made under the AGPLv3 and that your failure to recognize this is unjustified and bad faith. Please fully implement the AGPLv3 license.

**Comment by nadimkobeissi** [comment]
Joseph Bonneau has responded saying that he is okay with AGPLv3.

**Comment by ioerror** [comment]
I'm glad you have emailed everyone, thanks for doing so. I look forward to each person involved agreeing to the AGPL.

**Comment by nadimkobeissi** [comment]
Thank you, Jacob.

I admit to having been confrontational during this issue, and I apologize. However, I feel this could have been avoided had you properly licensed and forked the document from the beginning.

**Comment by ioerror** [comment]
Your confrontational behavior seems to be unavoidable regardless of my conduct. To tell me that I could have avoided your abuse by doing something you didn't even ask for in the original bug report is quite unreasonable. The emails you sent to numerous mailing lists, your tweets and your private emails to funders are also quite unreasonably hostile.

Once all of the co-authors agree to the license change, I'll make a note of the license being agreed upon by all involved parties.

**Comment by ioerror** [comment]
<i>I'll note to those reading along at home that the original ticket was titled "Plagiarism" when it was opened by @kaepora</i>

**Comment by nadimkobeissi** [comment]
I would not go so far as to call this abuse, and will ask you to please not slide over the lacking behaviour from your part. If I was in any way more confrontational than usual, it's probably due to your sour history with Cryptocat in general.

I also did not send emails to any mailing lists, other than replying to your email to otr-dev. I believe you should have properly licensed and attributed this repository before then and you made a mistake in not doing so.

**Comment by ioerror** [comment]
I'll call it abuse. Furthermore, please stop blaming me for your lack of self-control and kindly step off.

**Comment by nadimkobeissi** [comment]
@ioerror I'm moving this conversation to email for the time being.

**Comment by ioerror** [comment]
@kaepora - I'm not interested in an email thread that continues this madness.

Please respect my request to step off.

**Comment by abeluck** [comment]
I affirm that all my past contributions to the mpOTR specification document in question are licensed under the AGPLv3.

**Comment by hellais** [comment]
This is madness!

You are discussing over attribution issues related to a spec that is not even complete.

You know what would be awesome? That the spec were completed and implemented. Discussing on attribution of an unfinished specification is not fruitful.

Also, why do you even need to license a specification? Shouldn't it be in the public domain?

**Comment by abeluck** [comment]
+1

**Comment by ghost** [comment]
@hellais I agree completely.  This is pathetic to see an argument over something that is not even finished.  I can see where kaepora is coming from where he want(s)(ed) proper attribution for the spec as AGPLv3 and it's great the other co-authors are responding with their permission, but I think he has moved past the point of being petty.

This, in combination with another thread with him baselessly harassing ioerror, is obnoxious and needs to stop.  It's unprofessional to start fights (regardless of how utterly insignificant they are) on public boards, and a [[true]] apology is in order.

**Comment by nadimkobeissi** [comment]
@brantar I've apologized for being confrontational (a few messages above,) and would like to also apologize for the drama. I thought ioerror was acting in bad faith for it to take so much effort just to get him to properly attribute and license this document.

Now that most of the authors have acknowledged the proper attribution, I hope we can deal with this and move on.

**Comment by ghost** [comment]
I missed your "apology", but it didn't really matter because you went back and immediately blamed your behavior on ioerror's "sour" history with the Cryptocat project in general.  From what I have seen, the confrontations between you two have been over personal disputes where you are unwilling to compromise.

Again, I could see where you were coming from at the beginning of the argument, although I do not agree with how you handled the issue.  I agree and hope I'm able to collaborate with you in better circumstances.

**Comment by nadimkobeissi** [comment]
Fair enough. I don't think you're out of line in what you're saying.

**Comment by nadimkobeissi** [comment]
I've spent some time thinking about this and have to admit, I think this whole thing was 90% motivated by a personal fight I had with Jake earlier. What happened was that because of the fight, I got such a bad impression of Jacob that it made me mistrust anything he did in the future, which automatically knocked me on the defensive always whenever he made a decision somewhat related to anything I was working on.

I'm sorry I let that personal fight interfere with my judgment of Jacob on a purely professional issue. That's something I certainly should apologize for.
